src/dotnet/Chat.Service/ChatsBackend.cs

```diff
@@ -1663,18 +1665,28 @@ public virtual async Task OnPlaceRemoved(PlaceChangedEvent eventCommand, Cancell
     }
 
     // [EventHandler]
-    public virtual async Task OnChatChangedEvent(ChatChangedEvent eventCommand, CancellationToken cancellationToken)
+    public virtual Task OnChatChangedEvent(ChatChangedEvent eventCommand, CancellationToken cancellationToken)
     {
         if (Invalidation.IsActive)
-            return; // It just spawns other commands, so nothing to do here
+            return Task.CompletedTask; // It just spawns other commands, so nothing to do here
 
         var (chat, oldChat, kind) = eventCommand;
         if (kind == ChangeKind.Remove || chat.IsSummarized == false)
             // TODO(AK): Check if we need any events to stop flow
-            return;
+            return Task.CompletedTask;
+
+        return NeedsSummarization()
+            ? Flows.GetOrStart<ConversationSplitFlow>(chat.Id, cancellationToken)
+            : Task.CompletedTask;
 
-        if ((chat.IsSummarized ?? false) && oldChat?.IsSummarized == false)
-            await Flows.StartOrReset<ConversationSplitFlow>(chat.Id, null, nameof(OnChatChangedEvent), cancellationToken).ConfigureAwait(false);
+        bool NeedsSummarization()
+        {
+            if (kind is ChangeKind.Create)
```

Comment:
Why do we return true there regardless chat.IsSummarized

Reply:
fixed